File: Advanced/TemplaterServer/src/Aspose.cs
using Aspose.Words;
using System.IO;

namespace TemplaterServer.src
{
	public class Aspose : PdfConverter
	{
		public Stream Convert(Stream template, string extension)
		{
			var doc = new Document(template);
			var ms = new MemoryStream();
			doc.Save(ms, SaveFormat.Pdf);
			ms.Position = 0;
			return ms;
		}
	}
}

File: Advanced/TemplaterServer/src/main/java/hr/ngs/templater/server/Aspose.java
package hr.ngs.templater.server;

import com.aspose.words.Document;
import com.aspose.words.SaveFormat;

import java.io.ByteArrayInputStream;
import java.io.ByteArrayOutputStream;

public class Aspose implements PdfConverter {

	@Override
	public byte[] convert(byte[] template, String extension) throws Exception {
		ByteArrayInputStream input = new ByteArrayInputStream(template);
		ByteArrayOutputStream result = new ByteArrayOutputStream();
		Document doc = new Document(input);
		doc.save(result, SaveFormat.PDF);
		return result.toByteArray();
	}
}
